core/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -84,6 +84,8 @@
     public static final String DEFAULT_MAPPING = "_default_";
     public static final Setting<Long> INDEX_MAPPING_NESTED_FIELDS_LIMIT_SETTING =
         Setting.longSetting("index.mapping.nested_fields.limit", 50L, 0, Property.Dynamic, Property.IndexScope);
+    public static final Setting<Long> INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING =
+        Setting.longSetting("index.mapping.total_fields.limit", 0, 0, Property.Dynamic, Property.IndexScope);
```

Comment:
There seemed to be consensus about setting a default limit of 1000 in #13547. Let's do that?